### DISCUSSION TITLE
EPSS score integration

### DISCUSSION THREAD

**Discussion by ciscocollabguru** [aquasecurity/trivy#4543] [Q&A]
### Question

I'm wondering if there is the ability to utilize EPSS scores along with (or in lieu of) only CVSS?  Or if there are plans to begin to include these scores into the project?

### Target

None

### Scanner

Vulnerability

### Output Format

None

### Mode

None

**Comment by simar7** [comment]
Do you have more info on what the EPSS score is and who publishes them?

**Reply by ciscocollabguru** [reply]
You bet!  It was created by the same group that came up with CVSS to address the deficiencies of that scoring system. LunaSec has a good write-up on what it is (first link). FIRST has a public API to access the scores and it’s updated daily I believe (second link). 

https://www.lunasec.io/docs/blog/what-is-epss/
https://www.first.org/epss/api

**Comment by synack-security** [comment]
This would be great to include in the output of a scan.  Example query:
https://api.first.org/data/v1/epss?cve=CVE-2022-27225,CVE-2022-27223,CVE-2022-27218

**Comment by jthin** [comment]
Hello, 

As discussed this morning at Kubecon EU with @simar7, we need EPSS score with CVE score. It helps to review CVE score with another (security) perspective.
Qualys has this information: [more information](https://blog.qualys.com/qualys-insights/2022/10/10/in-depth-look-into-data-driven-science-behind-qualys-trurisk)

cc: @AnaisUrlichs

**Reply by simar7** [reply]
Hi thanks again for stopping by! We will take a look at it once we get back. 

cc @itaysk

**Reply by knqyf263** [reply]
EPSS is now available only in our commercial product.

**Reply by jthin** [reply]
Thank you for your response. Is there any plan to integrate it also for the open source version?

**Reply by knqyf263** [reply]
We currently have no plans to integrate EPSS into the OSS. More requests may change the plan, but we can't promise.

**Reply by EdgewareRoad** [reply]
As not in the OSS version of Trivy, my Trivy Summary tool (essentially a reporting tool and optimised for CI) now also includes EPSS, if that helps. https://github.com/EdgewareRoad/TrivySummary

**Reply by itaysk** [reply]
@EdgewareRoad thanks for sharing. would you want to add it to the trivy documentation under "ecosystem" section? feel free to open a pr. also, this might be a good case for a Trivy output plugin

**Reply by EdgewareRoad** [reply]
@itaysk Thanks for the suggestion. I'll take a look.

**Reply by pl0psec** [reply]
I've expanded my Trivy security findings report to include exploitability details for the vulnerabilities identified. I developed a tool that provides EPSS scores, and related Exploit-DB and Metasploit information for each CVE (feed is updated 4 times a daily). You can check it out at [plopsec.com](https://plopsec.com/). I hope this can be beneficial for others as well!

![image](https://github.com/user-attachments/assets/b27432f8-8555-48cd-acc5-b13ba3ccb654)

**Reply by itaysk** [reply]
nice! feel free to also make a PR to add to the Ecosystem page on Trivy's docs

**Comment by itaysk** [comment]
Summarizing this thread for the record: We will not implement EPSS scoring natively in Trivy as we consider it out of scope of the Trivy project. The scope and other core principles are documented [here](https://aquasecurity.github.io/trivy/v0.50/community/principles/#core-principles). Specifically, EPSS falls under "Exploitability of the vulnerability" as mentioned [here](https://github.com/aquasecurity/resources/blob/main/trivy-aqua.md#vulnerability-scanning).

**Comment by huornlmj** [comment]
The scan2html Trivy module also does this and also includes "has exploit" data.

**Comment by fatihtokus** [comment]
Hi @ciscocollabguru ,

I am the maintainer of the sacan2html Trivy module. Leaving its link here, in case of any new feature/bug please do not hesitate to get in touch with me: https://github.com/fatihtokus/scan2html

@huornlmj , many thanks for suggesting scan2html.

$ trivy scan2html -h

Usage: trivy scan2html [-h,--help] command target filename
  trivy scan2html <trivy [global flags] command [flags] target> --scan2html-flags [scan2html flags]
  trivy scan2html generate --scan2html-flags [scan2html flags]
  
Utility Commands
  generate    Generate a report from multiple json scan results
  help        Help about any command
  version     Print the version
  
Flags:
  -h, --help      Show usage.
  --output        Report name
  --report-title  Report Title
  --with-epss     Include EPSS data
  --with-exploits Include CISA Known Exploited Vulnerabilities
